sql: allow Java acceptance tests to work

madelynnblue opened the pull request:
Three changes make this possible.
1. Accept the VARCHAR OID and treat it like TEXT.
2. Support the limit field in execute commands as long as there aren't
   any leftover rows. If we find a driver or test case where this assumption
   doesn't hold (i.e., we will need to page through the results), we will
   add support for it later.
3. Silently ignore the EXTRA_FLOAT_DIGITS setting, which is sent by the
   JDBC driver during connection start.

Fixes #4035

<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.svg" height="40" alt="Review on Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/4496)

<!-- Reviewable:end -->

tamird commented:
LGTM

---

Reviewed 4 of 4 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[sql/pgwire/v3.go, line 551 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4496#-KAqHETM2AWlhY-ZfI4K:-KAqHETM2AWlhY-ZfI4L:198217920) ([raw file](https://github.com/cockroachdb/cockroach/blob/53aea61e5ed1676c31502e69ada51b40b3a8ffcd/sql/pgwire/v3.go#L551)):*
nit: any reason not to leave this as an int32? seems desirable to push the cast to the end user

---

*[sql/pgwire/v3.go, line 629 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4496#-KAqH7epm_QZ-hHf5YR3:-KAqH7epm_QZ-hHf5YR4:680878643) ([raw file](https://github.com/cockroachdb/cockroach/blob/53aea61e5ed1676c31502e69ada51b40b3a8ffcd/sql/pgwire/v3.go#L629)):*
s/%v/%d/

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4496#-:-KAqHR1OQPUhx8P9Eokd:676834233)*
<!-- Sent from Reviewable.io -->

madelynnblue committed:
sql: allow Java acceptance tests to work

Three changes make this possible.

1. Accept the VARCHAR OID and treat it like TEXT.

2. Support the limit field in execute commands as long as there aren't
any leftover rows. If we find a driver or test case where this assumption
doesn't hold (i.e., we will need to page through the results), we will
add support for it later.

3. Silently ignore the EXTRA_FLOAT_DIGITS setting, which is sent by the
JDBC driver during connection start.

Fixes #4035

tamird commented:
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[sql/pgwire/v3.go, line 617 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4496#-KAqXkGprGkKA5QdgjQp:-KAqXkGprGkKA5QdgjQq:1867591737) ([raw file](https://github.com/cockroachdb/cockroach/blob/86a54a122e3263961ca10a0f885db6029e3c700c/sql/pgwire/v3.go#L617)):*
i was thinking this'd be int32

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4496)*
<!-- Sent from Reviewable.io -->

mjibson commented:
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[sql/pgwire/v3.go, line 551 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4496#-KAqHETM2AWlhY-ZfI4K:-KAqW9GrCEkI50TXzrHd:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/53aea61e5ed1676c31502e69ada51b40b3a8ffcd/sql/pgwire/v3.go#L551)):*
Done.

---

*[sql/pgwire/v3.go, line 629 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4496#-KAqH7epm_QZ-hHf5YR3:-KAqWKGMZkOJU2xUr9WA:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/53aea61e5ed1676c31502e69ada51b40b3a8ffcd/sql/pgwire/v3.go#L629)):*
Done.

---

*[sql/pgwire/v3.go, line 617 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4496#-KAqXkGprGkKA5QdgjQp:-KAuTLqyidNHqUC1b6lj:340804932) ([raw file](https://github.com/cockroachdb/cockroach/blob/86a54a122e3263961ca10a0f885db6029e3c700c/sql/pgwire/v3.go#L617)):*
Ah. I wanted to avoid that because it has a silly cast in the logic. Opened #4518 if you'd prefer it the way you say, though.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4496)*
<!-- Sent from Reviewable.io -->
